File: 2015/src/day07/day07_test.c
#include "day07.h"
#include "testing.h"
#include "utils.h"

#include <stdint.h>

const test_entry* day07_tests[] = {
    TE("./data/example01.txt", "d07_p1_e01", one, 492),
    TE("./data/input.txt", "d07_p1_i", one, 46065),
    TE("./data/input.txt", "d07_p2_i", two, 14134),
    nullptr
};

int64_t day07_test_run(const test_entry* te) {
    char** input = file_to_array(te->file);
    int64_t r = (te->p == one) ? day07_part1((const char* const*)input) : day07_part2((const char* const*)input);
    free_array((void**)input);
    return r;
}

File: 2023/src/day07/day07_test.go
package day07

import (
	at "aoc2023/common/aoc_testing"
	"os"
	"testing"
)

var example01, input *string

func TestMain(m *testing.M) {
	example01 = at.MustRead("./data/example01.txt")
	input = at.MustRead("./data/input.txt")

	os.Exit(m.Run())
}

func TestD07p1(t *testing.T) {
	at.Run(t, part1, []at.Case[uint32]{
		{Name: "d07_p1_e01", Input: example01, Expected: 6440},
		{Name: "d07_p1_i", Input: input, Expected: 250058342},
	})
}

func TestD07p2(t *testing.T) {
	at.Run(t, part2, []at.Case[uint32]{
		{Name: "d07_p2_e01", Input: example01, Expected: 5905},
		{Name: "d07_p2_i", Input: input, Expected: 250506580},
	})
}
